### PR TITLE
Ee uart comms

### PR DESCRIPTION
**Pull request by EricEaton99**
## Kelpie Robotics - Underwater Microcontroller
Wrote PICommsDriver and Manager

### ⛔ Potential Risks
- Not tested on full UART since my console would only let me send 2 characters at a time.

### CODE REVIEW

File: kelpie-f446/Core/Modules/InterCommsModule.h [hunk 1]
```diff
@@ -8,6 +8,18 @@
 #ifndef MODULES_INTERCOMMSMODULE_H_
 #define MODULES_INTERCOMMSMODULE_H_
 
+#include "stm32f4xx_hal.h"
+#include <stdint.h>
+#include "UserTypes.h"
 
+typedef struct ICommsMsg_t{
+	uint8_t msgId;
```
Comment (mingyeeee) [inline kelpie-f446/Core/Modules/InterCommsModule.h]: uint16_t since 4 char hex is 16 bits

File: kelpie-f446/Core/Modules/InterCommsModule.h [hunk 2]
```diff
@@ -8,6 +8,18 @@
...
+	uint8_t dataLen;
```
Comment (mingyeeee) [inline kelpie-f446/Core/Modules/InterCommsModule.h]: uint16_t same reason

File: kelpie-f446/Core/UserDrivers/PiCommsDriver.c [hunk 3]
```diff
@@ -2,34 +2,49 @@
...
+PRIVATE PiCommsMessage_t PiComms_rxBufferToMessage();
 PRIVATE void PiCommsQueue_init(PiCommsQueue_t * q);
-PRIVATE uint8_t PiCommsQueue_dequeue(PiCommsQueue_t * q);
-PRIVATE void PiCommsQueue_enqueue(PiCommsQueue_t * q, uint8_t value);
+PRIVATE PiCommsMessage_t PiCommsQueue_dequeue(PiCommsQueue_t * q);
+PRIVATE void PiCommsQueue_enqueue(PiCommsQueue_t * q, PiCommsMessage_t value);
 
 PUBLIC void PiComms_Init(){
-	HAL_UART_Receive_IT(&huart4, piComms_rxBuffer, 1);
+	piComms_rxBuffer_index = piComms_rxBuffer = calloc(RX_BUFFER_SIZE, sizeof(uint8_t));
```
Comment (mingyeeee) [inline kelpie-f446/Core/UserDrivers/PiCommsDriver.c]: is calloc necessary? don't want any memory leaks by accident

File: kelpie-f446/Core/Modules/InterCommsModule.h (hunk 2 above)
Comment (EricEaton99) [reply kelpie-f446/Core/Modules/InterCommsModule.h]: I also made this fix in UserTypes.h since it also used uint8_t

**Pull request merged**